personhood/contract_spawner.go

```diff
@@ -0,0 +1,241 @@
+package personhood
+
+import (
+	"crypto/sha256"
+	"errors"
+
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/contracts"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/onet/v3/log"
+	"go.dedis.ch/protobuf"
+)
+
+// ContractSpawnerID denotes a contract that can spawn new instances.
+var ContractSpawnerID = "spawner"
+
+// SpawnerCoin defines which coin type is allowed to spawn new instances.
+var SpawnerCoin = byzcoin.NewInstanceID([]byte("SpawnerCoin"))
+
+// ContractSpawnerFromBytes returns a Spawner instance from a slice of bytes.
+func ContractSpawnerFromBytes(in []byte) (byzcoin.Contract, error) {
+	c := &ContractSpawner{}
+	err := protobuf.Decode(in, &c.SpawnerStruct)
+	if err != nil {
+		return nil, errors.New("couldn't unmarshal instance data: " + err.Error())
+	}
+	return c, nil
+}
+
+// ContractSpawner embeds the BasicContract.
+type ContractSpawner struct {
+	byzcoin.BasicContract
+	SpawnerStruct
+}
+
+// VerifyInstruction allows non-darc-verified calls for instructions that send coins.
+func (c ContractSpawner) VerifyInstruction(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instruction, ctxHash []byte) error {
+	if inst.GetType() != byzcoin.SpawnType {
+		if err := inst.Verify(rst, ctxHash); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Spawn creates a new spawner. Depending on what the client wants to spawn, this method will check the
```

Comment:
Where do the coins go? Are they just burned? Is that on purpose? Documented someplace?

Reply:
Added comment:
```
// Currently the coins are simply burned and will never be seen again. In future versions, the coins will
// be part of the mining reward for the nodes participating in the consensus.
```